elfpy/markets/hyperdrive/hyperdrive_market.py

```diff
@@ -437,20 +445,18 @@ def open_long(
         # apply deltas
         self.market_state.apply_delta(market_deltas)
         agent_wallet.update(agent_deltas)
-        checkpoint_time = self.latest_checkpoint_time
         # create/update the checkpoint
-        self.apply_checkpoint(checkpoint_time, self.market_state.share_price)
-        self.market_state.checkpoints[checkpoint_time].long_base_volume += market_deltas.long_base_volume
-        self.market_state.total_supply_longs[checkpoint_time] += market_deltas.longs_outstanding
+        self.apply_checkpoint(self.latest_checkpoint_time, self.market_state.share_price)
```

Comment:
should be at the top of open_long to mirror solidity

Reply:
If I put it at the top I fail tests.

Reply:
yeah it should be at the top

Reply:
done